### PR TITLE
Add README for mocks subpackage

### PR DESCRIPTION
**Pull request by wvanbergen**
README for the mocks subpackage

### CODE REVIEW

**Comment by wvanbergen** [comment]
@Shopify/kafka

File: mocks/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# sarama/mocks
+
+The `mocks` subpackage includes mock implementations that implement the interfaces of the major sarama types.
+You can use them to test your sarama applications using dependency injection.
+
+The following mock objects are available:
+
+- [Consumer](http://godoc.org/github.com/Shopify/sarama/mocks#Consumer) and PartitionConsumer
```
Comment (eapache) [inline mocks/README.md]: why no link for PartitionConsumer?

File: mocks/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+- [AsyncProducer](http://godoc.org/github.com/Shopify/sarama/mocks#AsyncProducer)
+- [SyncProducer](http://godoc.org/github.com/Shopify/sarama/mocks#SyncProducer)
```
Comment (eapache) [inline mocks/README.md]: https-all-the-things please

**Comment by eapache** [comment]
two nits, then :shipit:

**Pull request merged**